src/core/lib/iomgr/ev_posix.c

```diff
@@ -62,7 +63,9 @@ typedef struct {
 } event_engine_factory;
 
 static const event_engine_factory g_factories[] = {
-    {"poll", grpc_init_poll_posix}, {"legacy", grpc_init_poll_and_epoll_posix},
+    {"poll", grpc_init_poll_posix},
+    {"epoll", grpc_init_epoll_linux},
```

Comment:
epoll wants to be element 0 in this array (it should be preferred if it's available)